tests/puppeteer/campaigns/tools/linkchecker.js

```diff
@@ -0,0 +1,129 @@
+require('module-alias/register');
+// Using chai
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const files = require('@utils/files');
+const urlsList = require('./urls.js');
+
+const LOG_PASSED = process.env.LOG_PASSED || false;
+
+
+// Report dir and file
+const reportPath = 'campaigns/tools/reports';
+let filename;
+
+const output = {
+  startDate: new Date().toISOString(),
+  endDate: null,
+  pages: [],
+};
+let outputEntry = {
+  name: '',
+  url: '',
+  passed: [],
+  failed: [],
+  jsError: [],
+};
+let requestError = false;
+let requestTextError = '';
+let javascriptError = false;
+let jsError = '';
+
+let page;
+let browser;
+
+describe('Crawl every page for defects and issues', async () => {
+  before(async () => {
+    await files.createDirectory(reportPath);
+    // Create report dir
+    filename = await files.generateReportFilename();
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    await page.setExtraHTTPHeaders({
+      'Accept-Language': 'fr-FR',
+    });
+
+    await page.setRequestInterception(true);
+    // intercepts requests
+    await page.on('request', (request) => {
+      request.continue();
+    });
+    // intercepts responses
+    await page.on('response', (response) => {
+      checkResponseStatus(
+        response.request().url(),
+        response.status().toString(),
+      );
+    });
+    // intercepts JS errors
+    await page.on('pageerror', (pageerr) => {
+      jsError = pageerr.toString();
+      javascriptError = true;
+      outputEntry.jsError.push({
+        error: jsError,
+      });
+    });
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+    output.endDate = new Date().toISOString();
+    await files.createFile(reportPath, `${filename}.json`, JSON.stringify(output));
+  });
+
+  urlsList.forEach((section) => {
+    describe(`${section.name} - ${section.description}`, async () => {
+      section.urls.forEach((pageToCrawl, index) => {
+        it(`Crawling ${pageToCrawl.name} (${index + 1}/${section.urls.length})`, async () => {
+          /* eslint-disable no-param-reassign */
+          pageToCrawl.url = `${section.urlPrefix}${pageToCrawl.url}`
+            .replace('URL_BO', global.BO.URL)
```

Comment:
I put this this way because I didn't have access to `URL_BO` and `URL_FO` in `urls.js`. Maybe we could require in `urls.js` so that we don't have to make this replacement here ?

Reply:
I added global.BO.URL / global.FO.URL in all Urls